src/protocol/OpenConnectionRequest2.php

```diff
@@ -17,23 +17,34 @@
 namespace raklib\protocol;
 
 use raklib\utils\InternetAddress;
+use raklib\utils\Cookie;
 
 class OpenConnectionRequest2 extends OfflineMessage{
 	public static $ID = MessageIdentifiers::ID_OPEN_CONNECTION_REQUEST_2;
 
+	public int $cookie = 0;
 	public int $clientID;
+	public bool $clientSupportsSecurity = false; // Always false for the vanilla client.
 	public InternetAddress $serverAddress;
 	public int $mtuSize;
 
 	protected function encodePayload(PacketSerializer $out) : void{
 		$this->writeMagic($out);
+		if (Cookie::$serverHasSecurity) {
```

Comment:
I do not like this static usage

Reply:
encode() converts a packet class fields into binary, not to transfer server logic into binary. Therefore it shouldn't be reading any global settings, even if we were to make it static to begin with.